pkg/format/format.go

```diff
@@ -21,36 +20,106 @@ const (
 )
 
 var (
-	labelVar  string
-	fsTypeVar string
-	drives    map[string]bool
-	driveKeys []string
+	labelVar   string
+	fsTypeVar  string
+	forceVar   bool
+	verboseVar bool
+	drives     map[string]bool
+	driveKeys  []string
 )
 
+func hasPartitions(d string) bool {
+	err := exec.Command("sfdisk", "-d", d).Run()
+	return err == nil
+}
+
+func isEmptyDevice(d string) (bool, error) {
+	// default result
+	isEmpty := false
+
+	if verboseVar {
+		log.Printf("Checking if %s is empty", d)
+	}
+
+	out, err := exec.Command("blkid", d).Output()
+	if err == nil {
+		log.Printf("%s has content. blkid returned: %s", d, out)
+		// there is content, so exit early
+		return false, nil
+	}
+
+	if exiterr, ok := err.(*exec.ExitError); ok {
+		if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+			// blkid exitcode 2 (from the non-busybox version) signifies the block device has no detectable content signatures
+			if status.ExitStatus() == 2 {
+				if verboseVar {
+					log.Printf("blkid did not find any existing content on %s.", d)
```

Comment:
why is this conditional on `verboseVar` but other output, like above, is not?

Reply:
@rn, which other line are you referring to?

Reply:
@rn This line is the verbose output, without a failure condition. The next one is a failure condition.

`blkid` exitcodes:
- `0` = content exists
- `2` = content does not exist
- other = error

Reply:
See below -

Reply:
NM, just realised that you unconditionally print a message if, for whatever reason, the disk is not formatted and this case is not one of them, so all good

Reply:
The above isn't an error case. The one below is an error case. I only log verbose/debugging if `-verbose` is enabled, other than if it's an error (such as below)